src/MudletInstanceCoordinator.cpp

```diff
@@ -100,31 +122,61 @@ void MudletInstanceCoordinator::handleReadyRead()
     // Receive package paths and add them the install queue.
     QByteArray data = socket->readAll();
     const QStringList packagePaths = QString::fromUtf8(data).split(QChar::LineFeed, Qt::SkipEmptyParts);
-    mQueuedPackagePaths << packagePaths;
+    mQueuedUris << packagePaths;
     mMutex.unlock();
 
-    installPackagesLocally();
+    openUrisLocally();
 }
 
-// Find the active host and install queued packages to it
-void MudletInstanceCoordinator::installPackagesLocally()
+// Try to open queued URIs
+// This is called when:
+// - after a profile has been opened
+// - after another instance of Mudlet has transmitted a list of URIs to this instance
+void MudletInstanceCoordinator::openUrisLocally()
 {
+    mMutex.lock();
     QTimer::singleShot(0, this, [this]() {
         mudlet* mudletApp = mudlet::self();
         Q_ASSERT(mudletApp);
-        Host* activeHost = mudletApp->getActiveHost();
-        if (activeHost) {
-            installPackagesToHost(activeHost);
-        } else {
-            mudletApp->slot_showConnectionDialog();
+
+        QStringList skippedUris;
+
+        // Process queued URIs until we have to wait for a profile to be selected or loaded
+        while (!mQueuedUris.isEmpty()) {
```

Comment:
I don't think so. This function needs to unlock the mutex and it needs to remove items from the list in the specific way.

Reply:
I can change it to something like `for (int i = mQueuedUris.length() - 1; i >= 0; i--)` to avoid while, but I don't think a ranged for will work since the loop needs to remove items from the list during the loop.

Reply:
Sure!